ads/google/doubleclick.js

```diff
@@ -265,3 +244,51 @@ function doubleClickWithGlade(global, data, gladeExperiment) {
 function getCorrelator(global) {
   return makeCorrelator(global.context.clientId, global.context.pageViewId);
 }
+
+function centerAd() {
+  setStyles(dev().assertElement(global.document.querySelector('#c')), {
+    top: '50%',
+    left: '50%',
+    bottom: '',
+    right: '',
+    transform: 'translate(-50%, -50%)',
+  });
+}
+
+/**
+ * @param {Object} data
+ * @return {!string}
+ */
+export function selectGptExperiment(data) {
+  const fileExperimentConfig = {
+    21060540: 'gpt_sf_a.js',
+    21060541: 'gpt_sf_b.js',
+  };
+  // Note that reduce will return the first item that matches but it is
+  // expected that only one of the experiment ids will be present.
+  let expFilename;
+  (data['experimentId'] || '').split(',').forEach(
+    val => expFilename = expFilename || fileExperimentConfig[val]);
+  return expFilename;
+}
+
+export function writeAdScript(data, gptFilename) {
+  const url =
+  `https://www.googletagservices.com/tag/js/${gptFilename || 'gpt.js'}`;
+  if (gptFilename || data.useSameDomainRenderingUntilDeprecated != undefined
```

Comment:
you have doubleClickWithGpt twice within this function, you could combine into one lovely if clause:

if (gptFilename || data.useSameDomainRenderingUntilDeprecated != undefined || data.multiSize || ((global.context.location.href.indexOf('google_glade=0') > 0 || Math.random < 0.1) && global.context.location.href.indexOf('google_glade=1') < 0)) {
   doubleClickWithGpt(global, data, GladeExperiment.GLADE_CONTROL, url);
} else {
   ... glade code
}